src/Aspire.Hosting.Azure/Extensions/AzureSqlExtensions.cs

```diff
@@ -65,6 +66,67 @@ public static IResourceBuilder<SqlServerServerResource> AsAzureSqlDatabase(this
         return builder;
     }
 
+    /// <summary>
+    /// Configures SQL Server resource to be deployed as Azure SQL Database (server).
+    /// </summary>
+    /// <param name="builder">The builder for the SQL Server resource.</param>
+    /// <param name="configureResource">Callback to customize the Azure resources that will be provisioned in Azure.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<SqlServerServerResource> AsAzureSqlDatabaseConstruct(this IResourceBuilder<SqlServerServerResource> builder, Action<ResourceModuleConstruct, SqlServer, IEnumerable<SqlDatabase>>? configureResource = null)
+    {
+        var configureConstruct = (ResourceModuleConstruct construct) =>
+        {
+            var sqlServer = new SqlServer(construct, builder.Resource.Name);
+
+            sqlServer.AssignProperty(x => x.Administrators.AdministratorType, "'ActiveDirectory'");
+            sqlServer.AssignProperty(x => x.Administrators.IsAzureADOnlyAuthenticationEnabled, "true");
+            sqlServer.AssignParameter(x => x.Administrators.Sid, construct.PrincipalIdParameter);
+            sqlServer.AssignParameter(x => x.Administrators.Login, construct.PrincipalName);
+            sqlServer.AssignParameter(x => x.Administrators.PrincipalType, construct.PrincipalTypeParameter);
+            sqlServer.AssignProperty(x => x.Administrators.TenantId, "subscription().tenantId");
+
+            var sqlFirewall = new SqlFirewallRule(construct);
+            sqlFirewall.AssignProperty(x => x.StartIPAddress, "'0.0.0.0'");
+            sqlFirewall.AssignProperty(x => x.EndIPAddress, "'255.255.255.255'");
+
+            List<SqlDatabase> sqlDatabases = new List<SqlDatabase>();
+            foreach (var databaseNames in builder.Resource.Databases)
+            {
+                var databaseName = databaseNames.Value;
+                var sqlDatabase = new SqlDatabase(construct, sqlServer, databaseName);
+                sqlDatabase.AssignProperty(x => x.Location, "location"); // HACK
```

Comment:
Why is this needed? Is it because this isn't a child resource of the server?

Reply:
Arm requires it. It would be good to do automatically.

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-net/pull/42341